pkg/api/v1alpha1/edgeconnect/convert_from_test.go

```diff
@@ -0,0 +1,297 @@
+		Namespace:                  "c",
+		SelfLink:                   "d",
+		UID:                        "e",
+		ResourceVersion:            "f",
+		Generation:                 1,
+		CreationTimestamp:          metav1.Time{Time: time.Now()},
+		DeletionTimestamp:          &metav1.Time{Time: time.Now()},
+		DeletionGracePeriodSeconds: &deletionGracePeriodSeconds,
+		Labels: map[string]string{
+			"a": "b",
+		},
+		Annotations: map[string]string{
+			"c": "d",
+		},
+		OwnerReferences: nil,
+		Finalizers:      nil,
+		ManagedFields:   nil,
+	}
+}
+
+func getV1beta1Spec() edgeconnect.EdgeConnectSpec {
+	replicas := int32(1)
+
+	return edgeconnect.EdgeConnectSpec{
+		Annotations: map[string]string{
+			"a": "b",
+		},
+		Labels: map[string]string{
+			"c": "d",
+		},
+		Replicas:     &replicas,
+		NodeSelector: nil,
+		KubernetesAutomation: &edgeconnect.KubernetesAutomationSpec{
+			Enabled: true,
+		},
+		Proxy: &edgeconnect.ProxySpec{
+			Host:    "e",
+			NoProxy: "f",
+			AuthRef: "g",
+			Port:    1,
+		},
+		ImageRef: edgeconnect.ImageRefSpec{
+			Repository: "h",
+			Tag:        "i",
+		},
+		ApiServer: "j",
+		HostRestrictions: []string{
+			"k",
+			"l",
+		},
+		CustomPullSecret:   "m",
+		CaCertsRef:         "n",
+		ServiceAccountName: "o",
+		OAuth: edgeconnect.OAuthSpec{
+			ClientSecret: "p",
+			Endpoint:     "q",
+			Resource:     "r",
+			Provisioner:  true,
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU: resource.MustParse("1"),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU: resource.MustParse("2"),
+			},
+			Claims: []corev1.ResourceClaim{
+				{
+					Name: "s",
+				},
+			},
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name:  "t",
+				Value: "u",
+			},
+		},
+		Tolerations: []corev1.Toleration{
+			{
+				Key:   "v",
+				Value: "w",
+			},
+		},
+		TopologySpreadConstraints: []corev1.TopologySpreadConstraint{
+			{
+				LabelSelector: &metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{
+							Key: "x",
+						},
+					},
+				},
+			},
+		},
+		HostPatterns: []string{
+			"y",
+		},
+		AutoUpdate: true,
+	}
+}
+
+func getV1beta1Status() edgeconnect.EdgeConnectStatus {
+	return edgeconnect.EdgeConnectStatus{
+		Conditions: []metav1.Condition{
+			{
+				Type:   "Ready",
+				Status: metav1.ConditionFalse,
+			},
+			{
+				Type:   "NotReady",
+				Status: metav1.ConditionTrue,
+			},
+		},
+		KubeSystemUID:    "a",
+		DeploymentPhase:  status.Running,
+		UpdatedTimestamp: metav1.Time{Time: time.Now()},
+		Version: status.VersionStatus{
+			LastProbeTimestamp: &metav1.Time{Time: time.Now()},
+			Source:             "a",
+			ImageID:            "b",
+			Version:            "c",
+			Type:               "d",
+		},
+	}
+}
+
+func fromAreSpecsEqual(src *edgeconnect.EdgeConnectSpec, dst *EdgeConnectSpec) bool { //nolint:revive
+	if !reflect.DeepEqual(src.Annotations, dst.Annotations) {
```

Comment:
better to write it differently, path t testing.T and then
```
if !reflect.DeepEqual(src.Annotations, dst.Annotations) {
    t.Errorf("src annontations not '%s'. got=%s", src.Annotations, dst.Annotations)
    return false
```
it will help to easily find problems in future.